src/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -94,5 +97,38 @@ public long ExtendedExpiresIn
 
         public DateTimeOffset AccessTokenExpiresOn { get; private set; }
         public DateTimeOffset AccessTokenExtendedExpiresOn { get; private set; }
+
+        public string Authority { get; private set; }
+
+        internal MsalTokenResponse CreateFromBrokerResponse(Dictionary<string, string> responseDictionary)
+        {
+            if (responseDictionary.ContainsKey(ErrorDescription))
+            {
+                return new MsalTokenResponse
+                {
+                    Error = responseDictionary[Error],
+                    ErrorDescription = responseDictionary[ErrorDescription]
+                };
+            }
+
+            return new MsalTokenResponse
+            {
+                Authority = responseDictionary.ContainsKey("authority")
+                    ? AppConfig.AuthorityInfo.CanonicalizeAuthorityUri(CoreHelpers.UrlDecode(responseDictionary["authority"]))
+                    : null,
+                AccessToken = responseDictionary["access_token"],
+                RefreshToken = responseDictionary.ContainsKey("refresh_token")
```

Comment:
I am surprised to see this here - do we ever get an RT from the broker? In ADAL the RT always sits on the broker.